### PR TITLE
NetBox 4.x Full PR

### LINKED ISSUE #18: Specify that it's necessary to migrate as part of installation
The installation instructions omit the necessary step of running `manage.py migrate` after installing the plugin. It's an understandable oversight, but for users with little or no plugin experience it's a key piece of information that ought to be included.

### LINKED ISSUE #22: Broken database migrations during upgrade to Netbox 4.0
The plugin does not actually reflect the new requirement in terms of importing `PluginConfig` of netbox v4.0. (see https://docs.netbox.dev/en/stable/plugins/development/migration-v4/)

```
Applying database migrations (python3 netbox/manage.py migrate)...
Traceback (most recent call last):
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/core/management/base.py", line 413, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/core/management/base.py", line 459, in execute
    output = self.handle(*args, **options)
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/core/management/base.py", line 104, in wrapper
    saved_locale = translation.get_language()
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/utils/translation/__init__.py", line 211, in get_language
    return _trans.get_language()
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/utils/translation/__init__.py", line 66, in __getattr__
    if settings.USE_I18N:
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/conf/__init__.py", line 89, in __getattr__
    self._setup(name)
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/conf/__init__.py", line 76, in _setup
    self._wrapped = Settings(settings_module)
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/django/conf/__init__.py", line 190, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/opt/git/netbox/netbox/netbox/settings.py", line 735, in <module>
    plugin = importlib.import_module(plugin_name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/opt/git/netbox/venv/lib/python3.10/site-packages/netbox_floorplan/__init__.py", line 1, in <module>
    from extras.plugins import PluginConfig
ImportError: cannot import name 'PluginConfig' from 'extras.plugins' (unknown location)
```

### LINKED ISSUE #24: Current supported Netbox 4.0 Version?
Hello!

I noticed thta some users are trying to install this plugin in their Netbox v4 environment. Could you please update the compatibility?

Thanks!

### PR DESCRIPTION
**Pull request by cruse1977**
* Full support for NetBox 4.0 (JS/template/python fixes)
* removes /admin and into navigation
* background images disabled as this functionality wasn't included
* version bump to 0.4.0

# Fixes

* Fixes #18 
* Fixes #22
* Fixes #24

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,29 +15,48 @@ A netbox plugin providing floorplan mapping capability for locations and sites
...
+| 3.6         | >= 0.3.2  |
+| >4.0.2      | 0.4.0     |
 
 ## Installing
 
 The plugin is available as a Python package in pypi and can be installed with pip  
 
 ```
-pip install netbox-floorplan-plugin
+source /opt/netbox/venv/bin/activate
```
Comment (natm) [inline README.md:33]: Installation specific, not all Netboxes have the virtualenv location

File: README.md [hunk 2]
```diff
@@ -15,29 +15,48 @@ A netbox plugin providing floorplan mapping capability for locations and sites
...
+cd /opt/netbox
+sudo ./venv/bin/python3 netbox/manage.py makemigrations netbox_floorplan_plugin
+sudo ./venv/bin/python3 netbox/manage.py migrate
+```
+
 Restart NetBox and add `netbox-floorplan-plugin` to your local_requirements.txt
 
 See [NetBox Documentation](https://docs.netbox.dev/en/stable/plugins/#installing-plugins) for details
 
+## Contributors ##
```
Comment (natm) [inline README.md:51]: We aren’t including this as standard

File: docker-develop/Dockerfile
Comment (natm) [inline docker-develop/Dockerfile:1]: We shouldn’t bundle a full docker environment, we can include notes or a docker compose local config to just install the plugin

**Review by natm — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -48,16 +48,6 @@ Restart NetBox and add `netbox-floorplan-plugin` to your local_requirements.txt
 
 See [NetBox Documentation](https://docs.netbox.dev/en/stable/plugins/#installing-plugins) for details
 
-## Contributors ##
```
Comment (jeffgdotorg) [inline README.md]: What's the rationale for removing this section?

File: README.md [hunk 4]
```diff
@@ -29,8 +29,8 @@ A netbox plugin providing floorplan mapping capability for locations and sites
 
 The plugin is available as a Python package in pypi and can be installed with pip  
 
+
 ```
-source /opt/netbox/venv/bin/activate
```
Comment (jeffgdotorg) [inline README.md]: I tend to favor leaving all necessary instructions in place to up the odds of success for someone installing this plugin as their first plugin. If the deletion is about the pathname being possibly invalid, I think we can go with `${NETBOX_HOME}/venv/bin/activate` or similar.

**Review by jeffgdotorg — COMMENTED** [review]
Thanks for the great work on this @cruse1977!

**Pull request merged**